### DISCUSSION TITLE
Separate configuration for hosted Renovate vs. Renovate CLI?

### DISCUSSION THREAD

**Discussion by janpio** [renovatebot/renovate#8644] [Request Help]
We are thinking about possibly using the Renovate CLI for _some_ of our dependencies where we have special requirements that the hosted renovate GitHub app can not give us (running more often, less cache time for Npm, triggering via Slack etc). 

Is there a way to have the Renovate CLI use a different configuration than in the `renovate.json` that is in the default branch of the repo?

**Comment by rarkins** [comment]
Unfortunately I can't think of any way to achieve this

**Comment by viceice** [comment]
The only thing which will maybe work is to use the `force` block in the self-hosted bot `config.js` to override all relevant settings

**Reply by divyenduz** [reply]
Thanks, is it possible to override `baseBranches` and `packageRules` in `config.js`?

**Reply by rarkins** [reply]
repository settings will override config.js settings, specifically:
- if baseBranches are configured in `renovate.json` then it will take precedence (assuming you don't `force` it)
- if packageRules are configured in `renovate.json` then they'll come *after* any rules in `config.js`

**Reply by divyenduz** [reply]
Thanks, that is helpful, how do I supply `baseBranches` and `pacakgeRules` in `config.js` though? I was trying to find that in the docs here https://docs.renovatebot.com/self-hosted-configuration/

**Reply by rarkins** [reply]
All user config is valid in self hosted config, although you should normally try to avoid it because it's like a black box to anyone trying to configure renovate.json

**Reply by divyenduz** [reply]
This doesn't seem to be working for me, here is an output where I try to override `baseBranches`, the configuration is read but is not respected. It still used the `renovate.json` file from the repository. 

```
$ renovate --automerge true --automerge-type pr --dependency-dashboard
 WARN: Config needs migrating
       "originalConfig": {
         "onboarding": true,
         "cacheDir": "/private/var/folders/l5/8t163c4j3j746d9svk672stw0000gn/T",
         "token": "***********",
         "repositories": ["prisma/renovate-test"],
         "requireConfig": true,
         "forceCli": true,
         "force": true,
         "baseBranches": ["test"]
       },
       "migratedConfig": {
         "onboarding": true,
         "cacheDir": "/private/var/folders/l5/8t163c4j3j746d9svk672stw0000gn/T",
         "token": "***********",
         "repositories": ["prisma/renovate-test"],
         "requireConfig": true,
         "forceCli": true,
         "force": {"enabled": true},
         "baseBranches": ["test"]
       }
 INFO: Repository started (repository=prisma/renovate-test)
       "renovateVersion": "24.44.0"
 INFO: Dependency extraction complete (repository=prisma/renovate-test)
       "baseBranch": "master",
       "stats": {
         "managers": {
           "github-actions": {"fileCount": 2, "depCount": 2},
           "npm": {"fileCount": 1, "depCount": 6}
         },
         "total": {"fileCount": 3, "depCount": 8}
       }
 INFO: Dependency extraction complete (repository=prisma/renovate-test)
       "baseBranch": "automated/dev",
       "stats": {
         "managers": {
           "github-actions": {"fileCount": 2, "depCount": 2},
           "npm": {"fileCount": 1, "depCount": 5}
         },
         "total": {"fileCount": 3, "depCount": 7}
       }
 INFO: Dependency extraction complete (repository=prisma/renovate-test)
       "baseBranch": "automated/latest",
       "stats": {
         "managers": {
           "github-actions": {"fileCount": 2, "depCount": 2},
           "npm": {"fileCount": 1, "depCount": 5}
         },
         "total": {"fileCount": 3, "depCount": 7}
       }
 WARN: Base branch does not exist - skipping (repository=prisma/renovate-test)
       "baseBranch": "automated/patch-dev"
 INFO: Repository finished (repository=prisma/renovate-test)
       "durationMs": 19756
```

**Reply by rarkins** [reply]
You need whatever you're forcing *inside* the `force: {}` object

**Reply by viceice** [reply]
try to use a `config.js` with

```js
module.exports = {
  force: {
    baseBranches: ["test"],
    // more forced config options ...
  }
};
```
